add files to nastaran (#1)
zamin4life/string-to-react-element: README.md

> This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).

Reviewer: __[HIGH]__  
This README.md file doesn't belong to this package. Update it.

# string-to-react-element 

# How does it work? 

> ### You can pass a text and a renderer array, the renderer array includes objects that contains a regex and a renderer function. this module replace matched words in main text with your renderer.

Reviewer: ```suggestion
You can pass a text and a an array containing objects that each contains a regex and a renderer function. This module replaces matched words in main text with your renderer function result based on your regexes .
```

# installation

> `npm install string-to-react-element@0.0.12`

Reviewer: ```
npm install --save string-to-react-element
```

# example

```
import React from 'react';
import parser from 'string-to-react-element'

> import {Link} from '@material-ui/core'

Reviewer: ```suggestion
import { Link } from 'path/to/Link.jsx'
```

const urlRegex = /(https?:\/\/[^\s]+)|(http?:\/\/[^\s]+)/;
const mentionRegex = /(\s)([@][\w_-]+)|((^())[@][\w_-]+)/;
const hashtagRegex = /(\s)([#][\w_-]+)|((^())[#][\w_-]+)/;
const whiteSpaceRegex = /(\s)/;
const symbolRegex = /([a-zA-Z0-9]|[-!$%^&*()_+|~=`{}\[\]:";'<>?,.\/])/;

const App =() => {

>   const text = '#hello slddksml@lkmdslkdm @lksdmcsd';

Reviewer: ```suggestion
  const text = '#hello string-to-react-element@zamin4life @zamin4life';
```

>   const renderer = () => (
>     [

Reviewer: ```suggestion
  const renderer = [
```

      {
        regex: mentionRegex,
        renderer: splitItem => (

>         /* eslint-disable-next-line jsx-a11y/anchor-is-valid */

Reviewer: __[HIGH]__  
Remove all eslint disables.

          <Link>
            {splitItem}
          </Link>
        ),
      },
      {
        regex: hashtagRegex,
        renderer: splitItem => (
          /* eslint-disable-next-line jsx-a11y/anchor-is-valid */
          <Link>
            {splitItem}
          </Link>
        ),
      },
      {
        regex: urlRegex,
        renderer: splitItem => (
          /* eslint-disable-next-line jsx-a11y/anchor-is-valid */
          <Link href={splitItem}>
            {splitItem}
          </Link>
        ),
      },
      {
        regex: whiteSpaceRegex,
        renderer: splitItem => (
          splitItem.split('').map((item) => {
            if (item === '\n') {
              return <br />;
            }
            if (item === ' ') {
              return <span> </span>;
            }
            if (item === '\t') {
              return <span>   </span>;
            }
            return item;
          })
        ),
      },
      {
        regex: symbolRegex,
        renderer: splitItem => splitItem,
      },
    ]
  )

>   return (
>     <>
>     {parser(text,renderer())}
>     </>
>   )

Reviewer: ```suggestion
  return parser(text, renderer);
```
